File: ClimbLog.Api/ClimbLog.Domain/Models/Entities/Sector.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ClimbLog.Domain.Models.Entities;
public class Sector
{
    public int Id { get; set; }
    public required string Name { get; set; }
    public decimal XStart { get; set; }
    public decimal XEnd { get; set; }
    public decimal YStart { get; set; }
    public decimal YEnd { get; set; }
    public int GymId { get; set; }
}

File: climb-log-flutter/lib/climasys/api/models/sector.dart
class Sector {
  final int id;
  final String name;
  final num xStart;
  final num xEnd;
  final num yStart;
  final num yEnd;

  Sector({
    required this.id,
    required this.name,
    required this.xStart,
    required this.xEnd,
    required this.yStart,
    required this.yEnd,
  });

  factory Sector.fromJson(Map<String, dynamic> json) {
    return Sector(
      id: json['id'],
      name: json['name'],
      xStart: json['xStart'],
      xEnd: json['xEnd'],
      yStart: json['yStart'],
      yEnd: json['yEnd'],
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'name': name,
    'xStart': xStart,
    'xEnd': xEnd,
    'yStart': yStart,
    'yEnd': yEnd,
  };
}